backend/app/store/image/bolt_store.go

```diff
@@ -0,0 +1,154 @@
+package image
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"io"
+	"io/ioutil"
+	"path"
+	"time"
+
+	bolt "github.com/coreos/bbolt"
+	log "github.com/go-pkgz/lgr"
+	"github.com/pkg/errors"
+)
+
+type Bolt struct {
+	fileName  string
+	db        *bolt.DB
+	MaxSize   int
+	MaxHeight int
+	MaxWidth  int
+}
+
+const imagesStagedBktName = "images_staged"
+const imagesBktName = "images"
+const insertTimeBktName = "insert_times"
+
+func NewBoltStorage(fileName string, maxSize int, maxHeight int, maxWidth int, options bolt.Options) (*Bolt, error) {
+	db, err := bolt.Open(fileName, 0600, &options)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to make boltdb for %s", fileName)
+	}
+
+	err = db.Update(func(tx *bolt.Tx) error {
+		if _, e := tx.CreateBucketIfNotExists([]byte(imagesBktName)); e != nil {
+			return errors.Wrapf(e, "failed to create top level bucket %s", imagesBktName)
+		}
+		if _, e := tx.CreateBucketIfNotExists([]byte(imagesStagedBktName)); e != nil {
+			return errors.Wrapf(e, "failed to create top level bucket %s", imagesStagedBktName)
+		}
+		if _, e := tx.CreateBucketIfNotExists([]byte(insertTimeBktName)); e != nil {
+			return errors.Wrapf(e, "failed to create top level bucket %s", insertTimeBktName)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to initialize boltdb db %q buckets", fileName)
+	}
+	return &Bolt{
+		db:        db,
+		fileName:  fileName,
+		MaxSize:   maxSize,
+		MaxHeight: maxHeight,
+		MaxWidth:  maxWidth,
+	}, nil
+}
+
+func (b *Bolt) Save(fileName string, userID string, r io.Reader) (id string, err error) {
+	data, err := readAndValidateImage(r, b.MaxSize)
+	if err != nil {
+		return "", errors.Wrapf(err, "can't load image %s", fileName)
+	}
+
+	data, _ = resize(data, b.MaxWidth, b.MaxHeight)
```

Comment:
is there a reason why resized status ignores? FS store does this:

```
	if resized { // resized also converted to png
		id = strings.TrimSuffix(id, filepath.Ext(id)) + ".png"
		dst = f.location(f.Staging, id)
	}

```

Reply:
FS store uses `resized` flag only to keep file extension consistent with actual type. Bolt do not store information about extension and I don't found reasons to use flag.

Reply:
makes sense